### PR TITLE
[3/3] Workaround type erasure problem

### LINKED ISSUE #2360: Failed to compile : "export 'Props' (reexported as 'AreaProps') was not found in './cartesian/Area'
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Steps to reproduce
1) update package.json from "recharts": "^2.0.0-beta.8" to "recharts": "^2.0.0".
2) npm build

### What is expected?
Build successfull 

### What is actually happening?
Creating an optimized production build...
Failed to compile.

./node_modules/recharts/es6/index.js
"export 'Props' (reexported as 'AreaProps') was not found in './cartesian/Area'

| Environment | Info |
|---|---|
| Recharts | v2.0.0 |
| React | 16.14.0 |
| System | Ubuntu 20.04 |
| Browser | Chrome |

---
I had the issue both on Linux and Windows. Re-updating to  "recharts": "^2.0.0-beta.8" worked on my windows instance (not on linux one)

### PR DESCRIPTION
**Pull request by avindra**
* This fixes #2360 where types are not fully erased
 * also name every default export in the source code
   - this is aligned with the new prettier default rule. see example below:
   - https://github.com/recharts/recharts/runs/1678829121

Note: please review/merge the other 2 PRs first. The other 2 should be merged first. ~There may be conflicts after merging, which I can take care of.~

EDIT: Rebased to resolve conflicts.

 * 1: #2361
 * 2: #2366

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,58 +1,102 @@
-export { default as Surface, Props as SurfaceProps } from './container/Surface';
-export { default as Layer, Props as LayerProps } from './container/Layer';
+// "export type" declarations are being used to workaround babel
+// issue  declarations are being used to workaround babel issues 11464 12578
+//
+// see https://github.com/babel/babel/issues/11464#issuecomment-617606898
```
Comment (avindra) [inline src/index.ts:4]: Here is the fix. I tested the build locally and the referenced types are now being erased as intended.

**Review by avindra — COMMENTED** [review]
Pointer to the core fix.

I still think the default -> named exports change is something desirable, but if someone disagrees I can back it out.

**Comment by avindra** [comment]
Rebased PR to resolve conflict.

Checking to see why CI is failing. The looks are green, but exit code is 1 for some reason:

```
Process completed with exit code 1.
```

**Comment by avindra** [comment]
Karma is not running any tests for some reason. When it detects 0 tests, it will return exit code 1.

```
Firefox 83.0 (Ubuntu 0.0.0): Executed 0 of 0 SUCCESS (0 secs / 0 secs)
```

This was working earlier, an example can be seen here:

https://github.com/avindra/recharts/runs/1624154224?check_suite_focus=true

**Comment by xile611** [comment]
@avindra  test also fails in my local env.

**Comment by avindra** [comment]
@xile611  Looks like [`karma-webpack`](https://github.com/ryanclark/karma-webpack/releases) is not ready. After downgrading to 4, the tests are able to run. CI is passing now too.

**Comment by avindra** [comment]
There are native ways to test React without enzyme now.

I would suggest migrating to Jest:

 * https://reactjs.org/docs/testing-recipes.html
 * https://jestjs.io

This combination can fully replace karma, mocha, chai, etc. Which will give the project more options as it won't be so dependent on karma tooling dependencies.

And the tests are already using a lot of conventions available in Jest, so porting it over wouldn't be too much of a headache.

**Comment by xile611** [comment]
> P.S., some thoughts on testing:
> 
> There are alternative ways to test React without enzyme.
> 
> I would suggest migrating to Jest, which is very common and recommended in docs:
> 
> * https://reactjs.org/docs/testing-recipes.html
> * https://jestjs.io
> 
> This combination can fully replace karma, mocha, chai, etc. Which will give the project more options as it won't be so dependent on karma tooling dependencies.
> 
> And the tests are already using a lot of conventions available in Jest, so porting it over wouldn't be too much of a headache.

Awesome! I Agree! Jest is easier to use now.

**Pull request merged**